### PR TITLE
post: cli 1.5

### PR DESCRIPTION
**Pull request by cexbrayat**
WIP: you don't have to review right now

The CLI will probably come out after ng 5 (there are still some outstanding issues not fixed).

### CODE REVIEW

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+## Support for Angular 5 and its new compiler
+
+This is probably the biggest feature of this release!
+As Angular 5 ships with an improved compiler (check out [our article about that](http://blog.ninja-squad.com/2017/11/02/what-is-new-angular-5/) if you missed it),
+the CLI can now use it (and will automatically if your project uses ng 5).
+It gives us faster builds,
+and even an AoT mode that starts to be usable in development (using the incremental rebuild).
+
+For example, on a medium-size application, where `ng serve` rebuilds in ~400ms,
```
Comment (jnizet) [inline _posts/2017-11-03-angular-cli-1.5.md]: medium-sized?

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+[Angular CLI 1.5.0](https://github.com/angular/angular-cli/releases/tag/v1.5.0) is out with some nice new features!
+
+If you want to upgrade to 1.5.0 without pain (or to any other version, BTW), I have created a Github project to help: [angular-cli-diff](https://github.com/cexbrayat/angular-cli-diff). Choose the version you're currently using (1.2.1 for example), and the target version (1.5.0 for example), and it gives you a diff of all files created by the CLI: [angular-cli-diff/compare/1.2.1…1.5.0](https://github.com/cexbrayat/angular-cli-diff/compare/1.2.1...1.5.0). You have no excuse for staying behind anymore!
+
+Let's see what new features we have!
+
+## Support for Angular 5 and its new compiler
+
+This is probably the biggest feature of this release!
+As Angular 5 ships with an improved compiler (check out [our article about that](http://blog.ninja-squad.com/2017/11/02/what-is-new-angular-5/) if you missed it),
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: absolute link without domain would be better: `http://blog.ninja-squad.com/2017/11/02/what-is-new-angular-5/` -> `/2017/11/02/what-is-new-angular-5/`

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+It gives us faster builds,
+and even an AoT mode that starts to be usable in development (using the incremental rebuild).
+
+For example, on a medium-size application, where `ng serve` rebuilds in ~400ms,
+you can see the difference between the versions if you use AoT compilation:
+
+- `ng serve --aot` with ng 4.4 and CLI 1.4: ~4000-8000ms
+- `ng serve --aot` with ng 5.0 and CLI 1.5: ~1800-3600ms
+
+As you may know, building with the AoT mode gives you all the usual TypeScript errors
```
Comment (jnizet) [inline _posts/2017-11-03-angular-cli-1.5.md]: As you may know, building with the AoT mode also compiles the templates, and thus detects errors at build time that you would only discover (hopefully) at runtime in JIT mode.

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+and the errors you may have in your templates.
+Which is pretty useful, but used to be too slow to be usable in development.
+Now it starts to be more bearable, even if it's still slower than JiT compilation,
+but the Angular team is working hard on this, and it starts to show
+(even if it is still too slow on big project).
```
Comment (jnizet) [inline _posts/2017-11-03-angular-cli-1.5.md]: projects

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+That means that your build command:
+
+    ng build --prod --build-optimizer
+
+can be simplified to:
+
+    ng build --prod
+
+once you'll have updated to the CLI 1.5 and Angular 5.
```
Comment (jnizet) [inline _posts/2017-11-03-angular-cli-1.5.md]: once you have upgraded

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+## ES2015 as a target
+
+The CLI now supports ES2015 as the target of the build (and not only ES5).
+If you are targeting only modern browsers, you can now choose to output ES2015 code
+as the result of your build.
+Just change the `target` of your `tsconfig.json` file and you're set!
+(fun fact: it slims down the bundles generated when I tried it, but I can't say it will always be the case...).
```
Comment (jnizet) [inline _posts/2017-11-03-angular-cli-1.5.md]: it slimmed down the size of the generated bundles

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+- `ng serve --aot` with ng 5.0 and CLI 1.5: ~1800-3600ms
+
+As you may know, building with the AoT mode gives you all the usual TypeScript errors
+and the errors you may have in your templates.
+Which is pretty useful, but used to be too slow to be usable in development.
+Now it starts to be more bearable, even if it's still slower than JiT compilation,
+but the Angular team is working hard on this, and it starts to show
+(even if it is still too slow on big project).
+
+By default, the `build-optimizer` plugin (which does a little bit of extra work on your generated code, like removing unneeded decorators, adding hints for dead code removals, etc) will now be applied to your build
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: `etc` -> `etc.` (with a final `.`), I think.

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+    import { registerLocaleData } from '@angular/common';
+    import localeFr from '@angular/common/locales/fr';
+
+    registerLocaleData(localeFr);
+
+The problem is that if you build your application in 5 different languages,
+you need to do this for each language, and you don't want to include every locale in every build 🤔.
+
+This is the cool feature introduced by the CLI 1.5: you don't have to do this yourself!
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: yourself -> by yourself, no?

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+The CLI will automatically add these lines of code, with the correct locale data,
+when you build your app (based on the locale you specified when you build):
+
+    ng build --aot --locale=fr
+
+When you are building or serving the app in JiT mode, the CLI will do the same trick:
+
+    ng serve --locale=fr
+
+But for your app to work without AoT, you'll have to set the `LOCALE_ID` manually
```
Comment (jnizet) [inline _posts/2017-11-03-angular-cli-1.5.md]: So it's able to register the right locale data in JIT mode, but not to set the locale? Strange.

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+(by a "man in the middle" type of attack).
+You don't have to do anything, and your application is safer!
+
+## TypeScript version mismatch
+
+Angular CLI warns you if you are using a version of TypeScript
+which is not the recommended one with your Angular version
+(as it can lead to unnecessary headaches).
+You can now deactivate the warning for your project (if you are sure about what you are doing of course 😅)
+with `ng set warnings.typescriptMismatch=false` (it was only globally previously).
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: "it was only globally previously" -> "it was previously only global", no?

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+## appRoot configurable
+
+The `appRoot` is now configurable in a CLI app,
+the default being `app` as it is right now.
+The main use of this feature is for people who want to have several app in the same `src` directory,
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: app -> apps

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+each with their own name.
+
+## Schematics template
+
+The new `schematics` tool used by the CLI under the hood to generate ... schematics (blueprints) of projects,
+is very generic and allows to create your own schematics.
+But how do you do that? To help with this question, the CLI team released a collection called `@schematics/schematics` containing a schematic for a schematic (that's a lot of schematic, I know).
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: You could mention "meta", for a joke.

File: _posts/2017-11-03-angular-cli-1.5.md [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+This sample demonstrates a few common features and is a good starting point
+if you want to try to build your own project templates.
+
+    yarn global add @angular-devkit/schematics @schematics/schematics
+    schematics @schematics/schematics:schematic --name my-custom-schematics
+
+## Nx schematics
+
+A new unofficial schematic has been released by nwrl.io.
+As I was saying in my last blog post, it didn't took long for the community to start producing these!
```
Comment (clacote) [inline _posts/2017-11-03-angular-cli-1.5.md]: link to last blog post

File: _posts/2017-11-03-angular-cli-1.5.md (hunk 9 above)
Comment (cexbrayat) [reply _posts/2017-11-03-angular-cli-1.5.md]: Yes, see https://github.com/angular/angular-cli/pull/7730#issuecomment-338894444

File: _posts/2017-11-03-angular-cli-1.5.md (hunk 8 above)
Comment (cexbrayat) [reply _posts/2017-11-03-angular-cli-1.5.md]: I think it works without by...

**Pull request merged**